p8e-api/src/test/kotlin/stream/EventStreamTest.kt

```diff
@@ -0,0 +1,180 @@
+package stream
+
+import com.nhaarman.mockitokotlin2.*
+import com.tinder.scarlet.Lifecycle
+import com.tinder.scarlet.ShutdownReason
+import com.tinder.scarlet.WebSocket
+import com.tinder.scarlet.lifecycle.LifecycleRegistry
+import io.p8e.util.base64decode
+import io.p8e.util.base64encode
+import io.provenance.engine.domain.*
+import io.provenance.engine.service.TransactionQueryService
+import io.provenance.engine.stream.EventStreamFactory
+import io.provenance.engine.stream.domain.*
+import io.reactivex.Flowable
+import io.reactivex.rxkotlin.toFlowable
+import org.junit.Before
+import org.junit.Test
+
+class EventStreamTest {
+    val query = "tm.event='NewBlock'"
+
+    lateinit var observer: EventStreamResponseObserver<EventBatch>
+
+    lateinit var transactionQueryService: TransactionQueryService
+
+    lateinit var lifecycle: LifecycleRegistry
+
+    lateinit var eventStreamService: EventStreamService
+
+    @Before
+    fun setUp() {
+        observer = mock()
+
+        transactionQueryService = mock()
+
+        lifecycle = mock()
+
+        eventStreamService = mock()
+        whenever(eventStreamService.observeWebSocketEvent()).thenReturn(listOf(mock<WebSocket.Event.OnConnectionOpened<WebSocket>>()).toFlowable())
+    }
+
+    fun buildEventStream(eventTypes: List<String> = listOf("scope_created", "scope_updated"), startHeight: Long = 100): EventStreamFactory.EventStream {
+        return EventStreamFactory.EventStream(eventTypes, startHeight, observer, lifecycle, eventStreamService, transactionQueryService)
+    }
+
+    @Test
+    fun `History is not streamed if startHeight is less than 0`() {
+        setLastBlockHeight(100)
+
+        buildEventStream(startHeight = -1).streamEvents()
+
+        verify(transactionQueryService, times(0)).blocksWithTransactions(any(), any())
```

Comment:
Could we assert the return value? and maybe pass in real numbers?

Reply:
well in this case I am asserting that it wasn't ever called, so I think we would want the any argument and there is no return to assert...

Reply:
sorry, i wasn't clear, but was speaking in general for this method call, i believe `blockWithTransactions` method returns a List of Longs, think we can assert as well that an empty list is returned or a long value? with non-any inputs